scripts/provision-group-check-users.js

```diff
@@ -0,0 +1,324 @@
+    .filter(Boolean);
+
+  if (existingRoles.length===0) return false;
+  if (existingRoles.length>1) return "INVALID_MULTIPLE_ROLES";
+  return existingRoles[0];
+}
+
+
+/**
+ * Given a proposed username/email pairing, cross-reference against the existing
+ * cognito pool. We primarily search by username, as that's the index of the pool.
+ *
+ */
+async function crossReferenceUsernameAndEmail(group, member, usersByEmail, usersByUsername) {
+  console.log()
+  console.log(`Membership request for user "${member.username}" (${member.email}). Proposed role: ${member.role}`)
+  const emailMatch = searchCognito(usersByEmail, member.email);
+  const userNameMatch = member.username && searchCognito(usersByUsername, member.username);
+
+  // Situation 1: Username match (either exact or with different casing)
+  if (userNameMatch && userNameMatch.candidates.length===1) {
+    const cognitoUsername = userNameMatch.candidates[0];
+    userNameMatch.exact ?
+      console.log(`\tOne exact username match found`) :
+      console.log(`\tOne username match found but the case differs: "${cognitoUsername}"\n\tACTION: change the username casing`);
+
+    const role = await getUserRole(group, cognitoUsername);
+    role ?
+      console.log(`\tUser currently has role ${role}`) :
+      console.log(`\tNo role currently associated with the user for this group`);
+
+    const associatedEmail = getEmailFromUser(usersByUsername[cognitoUsername]);
+    if (associatedEmail===member.email) {
+      console.log(`\tThe email exactly matches the one associated with the existing user`)
+    } else if (associatedEmail.toUpperCase()===member.email.toUpperCase()) {
+      console.log(`\tThe cognito email matches but with different case: "${associatedEmail}"`)
+      console.log(`\tACTION: change the casing of the email to match`)
+    } else {
+      console.log(`\tYour email wasn't the one associated with the cognito user`)
+      console.log(`\tACTION: change email to "${associatedEmail}"`)
+      if (emailMatch) {
+        const users = emailMatch.candidates.flatMap((e) => usersByEmail[e].map((u) => u.Username));
+        console.log(`\t(NOTE: your provided email, "${member.email}", was associated with other user(s): ${users.join(", ")}, so you may want to use a different username instead)`)
+      }
+    }
+    return;
+  }
+
+  // Situation 2: Multiple usernames.
+  if (userNameMatch && userNameMatch.candidates.length>1) {
+    console.log(`\tThere are multiple matching usernames which differ by case:`)
+    for (const cognitoUsername of userNameMatch.candidates) {
+      const exactMsg = cognitoUsername===member.username ? '(exact match)' : '(case differs)';
+      const associatedEmail = getEmailFromUser(usersByUsername[cognitoUsername]);
+      const emailMsg = cognitoUsername===member.email ?
+        '(exact match)' :
+        cognitoUsername.toUpperCase()===member.email.toUpperCase() ? 
+          '(case differs)' :
+          '(not a match with the provided email)';
```

Comment:
Bug: username compared against email

```suggestion
      const emailMsg = associatedEmail===member.email ?
        '(exact match)' :
        associatedEmail.toUpperCase()===member.email.toUpperCase() ? 
          '(case differs)' :
          '(not a match with the provided email)';
```